lua/nvim-treesitter/configs.lua

```diff
@@ -299,8 +299,8 @@ end
 -- @param bufnr the bufnr
 -- @param lang the language of the buffer
 function M.attach_module(mod_name, bufnr, lang)
-  local bufnr = bufnr or api.nvim_get_current_buf()
-  local lang = lang or parsers.get_buf_lang(bufnr)
```

Comment:
Why having a global here ?

Reply:
When you use the local keyword the variable will be redefined, and the langserver will give a warning. Since the variable comes from the function argument it is already local. It is like doing
```lua
local bufnr = <user arg> -- from the function parameter
bufnr = bufnr or api.nvim_get_current_buf() -- not global, mutating local
```

Reply:
Its kind of a small change, mostly just to make the language server not give out a warning.

Reply:
Oh yeah sorry for that, is see now, and that's better indeed